test/helper/e2e/actions/cloud/gcp.go

```diff
@@ -360,7 +361,15 @@ func (a *GCPAction) randomIP(subnet string) string {
 	}
 
 	for {
-		ipParts[3] = strconv.Itoa(rand.IntN(255))
+		const maxRandValue = 256
+
+		randNumberBig, err := rand.Int(rand.Reader, big.NewInt(maxRandValue))
+		if err != nil {
+			panic(fmt.Errorf("failed to generate a random number: %w", err))
+		}
+
+		randNumber := int(randNumberBig.Int64())
+		ipParts[3] = strconv.Itoa(randNumber)
```

Comment:
why the extra complexity here? can you elaborate?

Reply:
> test/helper/e2e/actions/cloud/gcp.go:363:29: G404: Use of weak random number generator (math/rand or math/rand/v2 instead of crypto/rand) (gosec)
>                 ipParts[3] = strconv.Itoa(crypto.rand(255))